test/Test/Database/LSMTree/Internal.hs

```diff
@@ -49,11 +37,6 @@ tests = testGroup "Test.Database.LSMTree.Internal" [
         , testProperty "twiceOpenSession" twiceOpenSession
         , testCase "sessionDirLayoutMismatch" sessionDirLayoutMismatch
         , testCase "sessionDirDoesNotExist" sessionDirDoesNotExist
-        , testProperty "prop_interimRestoreSessionUniqueRunNames"
-            prop_interimRestoreSessionUniqueRunNames
-        ]
-    , testGroup "Table" [
-          testProperty "prop_interimOpenSnapshot" prop_interimOpenSnapshot
```

Comment:
Do we want to add new tests in exchange? We still test snapshotting through the public API in the state machine and class-based tests, but we usually also have tests for the internals to catch issues more directly.

Reply:
`interimOpenSnapshot` does not have much value over the class-based tests and state machine tests we already have, but since we don't actually test the session that much I've re-introduced some of the testing stuff into the `newSession` and `restoreSession` properties